### PR TITLE
fix up minion start

### LINKED ISSUE #57794: [BUG] pidfile is removed on check
**Description**
This bug makes possible to start a few copies of minion because the PID file is removed during the process of startup of the 2nd minion.
This short video demonstrated it - http://youtu.be/Ac-IBu7qnLY

**Expected behavior**
Always detect a pif file and do not run.

**Versions Report**
Salt 3001.

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
Stops a Minion from starting if a minion is all ready running.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/57794

### Previous Behavior
Will make a 'pidfile' and put a PID in it. If the PID does not match the minion it will not start "A minion all ready running". The minion that failed to start will delete the file after a check "Problem!". 

### New Behavior
Rather than stopping failed minions from deleting the file.

I did this...
I write the pid, pid_name and pid_start_time to the pidfile. This gives the new process enough information to know if the old minion is dead or not. So now we can just do a single call to `claim_mantle_of_responsibility` which the process will claim responsibly if it can. It will return a bool letting us know if it was successful or not.

I did this because if the minion crashes the pidfile wont be deleted. This will stop minion from starting in the future. This will be a huge pain for people who are new to salt and for people who just kill the minion. 

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by cmcmarrow** [comment]
re-run full all

**Comment by cmcmarrow** [comment]
re-run full all

**Comment by cmcmarrow** [comment]
re-run full all

**Comment by cmcmarrow** [comment]
re-run full lint

**Comment by cmcmarrow** [comment]
re-run full windows

**Comment by cmcmarrow** [comment]
re-run full ubuntu

**Review by Akm0d — DISMISSED** [review]

File: salt/utils/process.py [hunk 1]
```diff
@@ -195,71 +196,113 @@ def notify_systemd():
...
+    Checks that no other live processes has this responsibility.
+    If claiming the mantle of responsibility was successful True will be returned.
+    file_name: str
+    Return: bool
+    """
+
+    # all OSs supported by salt has psutil
+    if not HAS_PSUTIL:
+        log.critical(
+            "Assuming no other Process has this responsibly! pidfile: {}".format(
```
Comment (waynew) [inline salt/utils/process.py:238]: responsibly or responsibility?

File: salt/utils/process.py [hunk 2]
```diff
@@ -195,71 +196,113 @@ def notify_systemd():
...
-    Return the pid from a pidfile as an integer
+    Sees who has the mantle of responsibility
+    file_name: str
+    Return: None or int
     """
+
+    # all OSs supported by salt has psutil
+    if not HAS_PSUTIL:
+        log.critical(
+            "Assuming no other Process has this responsibly! pidfile: {}".format(
```
Comment (waynew) [inline salt/utils/process.py:287]: same ? here - responsibility or responsibly?

File: salt/utils/process.py [hunk 3]
```diff
@@ -195,71 +196,113 @@ def notify_systemd():
...
     try:
-        with salt.utils.files.fopen(pidfile) as pdf:
-            pid = pdf.read().strip()
-        return int(pid)
-    except (OSError, IOError, TypeError, ValueError):
-        return -1
+        with salt.utils.files.fopen(file_name, "r") as file:
+            file_process_info = json.load(file)
+    except json.decoder.JSONDecodeError:
+        log.error("pidfile:{} is corrupted".format(file_name))
```
Comment (waynew) [inline salt/utils/process.py:298]: no space after `:` - messages should be consistent.

**Review by waynew — CHANGES_REQUESTED** [review]
Couple questions and at least a change to make.

File: salt/utils/process.py (hunk 1 above)
Comment (cmcmarrow) [reply salt/utils/process.py]: responsibility

File: salt/utils/process.py (hunk 3 above)
Comment (cmcmarrow) [reply salt/utils/process.py]: space

**Review by waynew — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by Akm0d — APPROVED** [review]

File: salt/utils/process.py [hunk 4]
```diff
@@ -195,71 +196,113 @@ def notify_systemd():
...
+
+def check_mantle_of_responsibility(file_name):
     """
-    Return the pid from a pidfile as an integer
+    Sees who has the mantle of responsibility
+    file_name: str
+    Return: None or int
     """
+
+    # all OSs supported by salt has psutil
```
Comment (rst0git) [inline salt/utils/process.py:284]: When salt is used inside a Docker container `psutil` may not be installed. In such case, the salt-minion is running but no pidfile is being created.

This change broke the virt integration tests in https://github.com/saltstack/salt/pull/57736 but, fortunately, installing the `python3-psutil` package in the container resolved this issue.

(cc @cbosdo )

**Pull request merged**